lib/src/main/java/graphql/nadel/validation/NadelNamespaceValidation.kt

```diff
@@ -1,18 +1,14 @@
 package graphql.nadel.validation
 
-import graphql.nadel.engine.util.operationTypes
-import graphql.nadel.engine.util.unwrapAll
-import graphql.nadel.util.NamespacedUtil.isNamespacedField
 import graphql.nadel.validation.NadelSchemaValidationError.NamespacedTypeMustBeObject
 import graphql.schema.GraphQLObjectType
 
-class NadelNamespaceValidation(
-) {
+class NadelNamespaceValidation {
     context(NadelValidationContext)
     fun validate(
         schemaElement: NadelServiceSchemaElement,
     ): NadelSchemaValidationResult {
-        if (!isNamespacedOperationType(typeName = schemaElement.overall.name)) {
+        if (schemaElement.overall.name !in namespaceTypeNames) {
```

Comment:
`NadelNamespaceValidation` is run for every type.

The old `isNamespacedOperationType` would iterate through all root fields, which was very slow.

This procomputes it and puts that information in the `NadelValidationContext` so that we only iterate through the root fields once.